pkg/istio-agent/agent.go

```diff
@@ -381,6 +381,14 @@ func (a *Agent) Run(ctx context.Context) (func(), error) {
 			if err := a.xdsProxy.initIstiodDialOptions(a); err != nil {
 				log.Warnf("Failed to init xds proxy dial options")
 			}
+
+			if s := a.secretCache; s != nil {
+				log.Info("ROOTCA changed, regenerating certs")
+				s.Reset()
+				a.getWorkloadCerts(s)
+				// Tirgger a push to envoy
+				s.OnSecretUpdate(security.WorkloadKeyCertResourceName)
```

Comment:
Could you explain what is happening with regards to updating the trust anchor here https://github.com/zirain/istio/blob/master/security/pkg/nodeagent/cache/secretcache.go#L308-L314? Is the goal of the above change to invalidate the cache in response to a root CA for XDS file update?

Reply:
During my test, workload cert will not be push to envoy without this line.

Reply:
This is concerning to me. 

1. XDS root and workload root are not the same in many cases; we should never couple the code for them
2. There is already code to dynamically update workload root via the "PCDS"

Reply:
My question is:
1. shouldn't istio retrieve workload root when xds root change?
2. shouldn't istio retrieve workload keycert when xds root or workload root change?